src/read/coff/section.rs

```diff
@@ -470,8 +470,21 @@ impl pe::ImageSectionHeader {
         &self,
         data: R,
     ) -> read::Result<&'data [pe::ImageRelocation]> {
-        let pointer = self.pointer_to_relocations.get(LE).into();
-        let number = self.number_of_relocations.get(LE).into();
+        let mut pointer = self.pointer_to_relocations.get(LE).into();
+        let mut number: usize = self.number_of_relocations.get(LE).into();
+        if number == u16::MAX.into()
+            && self.characteristics.get(LE) & pe::IMAGE_SCN_LNK_NRELOC_OVFL != 0
+        {
+            // Extended relocations. Read first relocation and read extended count & adjust
+            // relocations pointer.
+            let extended_relocation_info: &pe::ImageRelocation = data
+                .read_at(pointer)
+                .read_error("Invalid COFF relocation offset or number")
+                .expect("Invalid COFF relocation offset or number");
+            // Extended relocation info does not contribute to the count of sections.
+            number = extended_relocation_info.virtual_address.get(LE) as usize - 1;
```

Comment:
There's a potential underflow.

Reply:
Yes, we should return an error if the raw number is 0. https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#section-flags also says it is an error if there are fewer than 0xffff, but I'm not sure if we should be that strict when parsing.

Reply:
That is totally up to you. IIRC llvm-objdump handles extended relocations with less than 0xffff entries.